data/basics/userBalance_test.go

```diff
@@ -190,8 +194,11 @@ func TestEncodedAccountDataSize(t *testing.T) {
 		}
 		maxLocalState[maxKey] = maxValue
 	}
-
-	for appCreatorApps := 0; appCreatorApps < currentConsensusParams.MaxAppsCreated; appCreatorApps++ {
+	maxAppsCreate := currentConsensusParams.MaxAppsCreated
+	if maxAppsCreate == 0 {
+		maxAppsCreate = config.Consensus[protocol.ConsensusV30].MaxAppsCreated
```

Comment:
It's unclear to me what this test actually does in a world with unlimited assets.  On the one hand, setting maxAppsCreate to the v30 level means we're lying - the return maximum size is wildly wrong (there is no limit to the size of marshaling basics.AccountData) On the other hand, I don't know why the test is checking the size here. Presumably we will never marshal one of these things again?

Reply:
This unit test is still there as a "backing" for the `MaxEncodedAccountDataSize` constant. This constant is currently used in the catchpoint files, such as `encodedBalanceRecordV5`. In the future, we're going to stop using this constant, as we plan to refactor the catchpoint file structure. But we still need to support the old files, and so this test need to stay "as-is".

Reply:
That makes sense. If it's true that when proto.MaxAppsCreated == 0 that the size of the encoded account does not grow at all (I think that's true), then I'd rather see this unit test leave maxAppsCreate == 0 when the proto value == 0, rather than set it to an old value from v30.  It's easier to understand that creating apps does not affect the size, and therefore isn't done in the test, than it is to understand why the v30 value is used.

Perhaps the test should be that the encoded size does not grow at all when apps are created if proto.MaxAppsCreated == 0.